src/SFA.DAS.RequestApprenticeTraining.Database/Tables/ProviderResponse.sql

```diff
@@ -7,6 +7,7 @@
 	[AcknowledgedAt] DATETIME2 NULL,
 	[AcknowledgedBy] UNIQUEIDENTIFIER NULL,
 	[RespondedAt] DATETIME2 NOT NULL,
+	[RespondedBy] VARCHAR(256) NOT NULL,
```

Comment:
These 'By' fields should be the user who is doing the operation.

I looked in the claims and there is a 'sub' claim - I suspect that this is what could be used for looking up the users email and name from the DfeSignIn API - we have a spike on that coming up.

We should include RespondedBy as a UNIQUEIDENTIFIER  NOT NULL and additionally add ContactName as an NVARCHAR NOT NULL (potentially has non-english characters)

When adding a NOT NULL Fields will need to add a default for existing rows - you should name the default explicitly in both tables see EmployerRequest.RequestedAt for an example - in this case you will need to default the RespondedBy column to {0000-00000.........    i.e the empty guid and the ContactName to an empty string.

You would store the Sub claim in the RespondedBy

You would store the display name from the claims in the ContactName column for now - after the spike we may use the DfeSignIn API - or maybe defer it to the MVS+.

In MVS+ we may be looking up from the DfeSignIn API the other users belonging to the current provider and allowing the current users to choose one - hence the RespondedBy would be different users to the ContactName

Reply:
Updated